blue_bringup/scripts/calibrate_joint_offsets_new.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+
+import numpy as np
+
+
+def old(actuator_angles):
+
+    side = "right"
+    version = "2"
+
+    gr1 = 7.1875
+    gr2 = 8.2444852941
+
+    if side == 'left':
+        rot2 = np.pi / 2
+    elif side == 'right':
+        rot2 = -np.pi / 2
+
+    if version == '1':
+        rot1 = -2.189
+    elif version == '2':
+        rot1 = -2.310
+
+    ms = actuator_angles
+    a0 = ms[0]
+    a1 = ms[1] + rot1 * gr1 + rot2 * gr2
+    a2 = ms[2] + -rot1 * gr1 + rot2 * gr2
+    a3 = ms[3] + rot1 * gr1 - rot2 * gr2
+    a4 = ms[4] + -rot1 * gr1 - rot2 * gr2
+    a5 = ms[5] + -2.189 * gr1
+    a6 = ms[6] + 2.189 * gr1
+    actuators = [a0, a1, a2, a3, a4, a5, a6]
+
+    return np.array(actuators)
+
+
+def new(actuator_angles):
```

Comment:
@wuphilipp this is nowhere near done but can you glance over this function and check that it makes sense?

notably, it currently gives different results from the old implementation -- negating `roll_ratio` on line#49 gives us the "correct" results, but I'm having trouble wrapping my head around where this negative sign comes from. maybe it's equivalent?

Reply:
The negative sign is a result of the hardware interface differential definition. I'm not sure what ways is best to convay the most intuitive and understandable code. Maybe add the negative in `propagate_through_differential`?
```
            right_actuator_angle = lift_angle * lift_ratio - roll_angle * roll_ratio
            left_actuator_angle = -lift_angle * lift_ratio - roll_angle * roll_ratio
```

Reply:
I just verified that the above equation. See the differential transmission implementation here.
https://github.com/ros-controls/ros_control/blob/melodic-devel/transmission_interface/include/transmission_interface/differential_transmission.h#L304-L305

And the signs that we use here.
https://github.com/berkeleyopenarms/blue_core/blob/master/blue_hardware_interface/src/blue_kinematics.cpp#L64-L67

If you directly apply these, however, you get the negation of these equations. We want the applied calibrations to be reverse to "realign" the calibration position to the original position, which is where the negation comes in.

Reply:
ah -- I think my confusion was because I forgot we flipped the right/left motor order

this was because the v2 links were wired with the left motors first?

Reply:
yup!